libcontainer/libcontainer_fuzzer.go

```diff
@@ -0,0 +1,85 @@
+// +build gofuzz
+
+package libcontainer
+
+import (
+	"bytes"
+	"crypto/md5"
+	"encoding/hex"
+	"encoding/json"
+	"errors"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/opencontainers/runc/libcontainer/configs"
+)
+
+func checkRootInData(data []byte, dir string) error {
+	for i := 0; i < len(dir); i++ {
+		if !strings.Contains(string(data), dir[:i]) {
+			return errors.New(dir[:i] + " not found in " + string(data))
+		}
+	}
+	return nil
+}
+
+func FuzzStateApi(data []byte) int {
+	if len(data) < 3 {
+		return -1
+	}
+	if strings.Contains(string(data), "null") {
+		return -1
+	}
+	root, err := newTestRoot()
+	if err != nil {
+		return -1
+	}
+
+	// The path to our root dir MUST be found
+	// in our input
+	err = checkRootInData(data, root)
+	if err != nil {
+		return -1
+	}
+	var config *configs.Config
+	reader := bytes.NewReader(data)
+	if err := json.NewDecoder(reader).Decode(&config); err != nil {
```

Comment:
I still don't understand how `data` is guaranteed to be unmarshallable JSON

Reply:
It is not guaranteed. But in the case of this fuzzer that doesn't matter, since we are using the validator. Over time the fuzzer will generate data that is unmarshallable JSON, but every test case will at no time be guaranteed to be useful.
If this has consequences for the fuzzer, then I will adjust it.